crates/holochain/src/conductor/state.rs

```diff
@@ -26,12 +25,18 @@ pub struct ConductorState {
     pub active_apps: HashMap<AppId, Vec<InstalledCell>>,
     /// List of interfaces any UI can use to access zome functions.
     #[serde(default)]
-    pub app_interfaces: HashMap<AppInterfaceNick, AppInterfaceConfig>,
+    pub app_interfaces: HashMap<AppInterfaceId, AppInterfaceConfig>,
 }
 
-/// A friendly name used to refer to an App Interface.
-// TODO: is this needed?
-pub type AppInterfaceNick = String;
+/// A unique identifier used to refer to an App Interface internally.
+#[derive(Clone, Deserialize, Serialize, Default, Debug, Hash, PartialEq, Eq, derive_more::From)]
+pub struct AppInterfaceId(String);
```

Comment:
This might lead to weird bugs if the strings aren't unique?

Reply:
Yeah definitely. I guess it could be a bit better if it were an enum? Can we ensure uniqueness in the type system?